src/lib/slide-toggle/slide-toggle.ts

```diff
@@ -333,12 +333,14 @@ class SlideToggleRenderer {
 
   /** Updates the thumb containers position from the specified distance. */
   updateThumbPosition(distance: number) {
-    this.dragPercentage = this._getThumbPercentage(distance);
-    applyCssTransform(this._thumbEl, `translate3d(${this.dragPercentage}%, 0, 0)`);
+    this.dragPercentage = this._getDragPercentage(distance);
+    // Calculate the moved distance based on the thumb bar width.
+    let dragX = (this.dragPercentage / 100) * this._thumbBarWidth;
+    applyCssTransform(this._thumbEl, `translate3d(${dragX}px, 0, 0)`);
```

Comment:
I tried to use `ngStyle` but actually ran into more issues than advantages. 

The slide-toggle would need to trigger a new round of change-detection `onDragEnd`, so that the drag styles are reverted.

I feel like for the OnPush strategy it is more beneficial to use `applyTransform`.